### PR TITLE
Pass code block to fixer prompt, not the raw LLM response

### PR DESCRIPTION
**Pull request by DonggeLiu**
Passing the raw response can confuse LLM when it contains more text than the code block, which occurred on `Gemini 1.5`.
Also, `Gemini 1.5`'s response structure seems to be different from `code-bison-32k`.

This PR does two things:
1. Make the response parser compatible with both models.
2. Pass the code block in response to LLM fixer, not the raw response.

### CODE REVIEW

File: llm_toolkit/output_parser.py [hunk 1]
```diff
@@ -44,24 +44,33 @@ def parse_args() -> argparse.Namespace:
   return args
 
 
+def _parse_code_block_by_marker(lines: list[str], start_marker: str,
+                                end_marker: str) -> list[str]:
+  """Parses code block lines based on markers."""
+  block, in_block = [], False
```
Comment (oliverchang) [inline llm_toolkit/output_parser.py:50]: nit: just have two separate lines for these. it's clearer.

**Review by oliverchang — APPROVED** [review]
thanks!!

**Comment by DonggeLiu** [comment]
JOB: https://console.cloud.google.com/kubernetes/job/us-central1-c/llm-experiment/default/ofg-pr-259-dg-g15
REPORT: https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-05-09-259-dg-g15-comparison/index.html
BUCKET: https://console.cloud.google.com/storage/browser/oss-fuzz-gcb-experiment-run-logs/Result-reports/ofg-pr/2024-05-09-259-dg-g15-comparison
BUCKET GS: `gs://oss-fuzz-gcb-experiment-run-logs/Result-reports/ofg-pr/2024-05-09-259-dg-g15-comparison`

File: llm_toolkit/output_parser.py (hunk 1 above)
Comment (DonggeLiu) [reply llm_toolkit/output_parser.py]: Done, thanks for the quick review!

**Comment by DonggeLiu** [comment]
/gcbrun request_pr_exp.py -n dg-g15 -m vertex_ai_gemini-1-5 -f

**Comment by DonggeLiu** [comment]
/gcbrun request_pr_exp.py -n dg-g15 -m vertex_ai_gemini-1-5 -f

**Comment by DonggeLiu** [comment]
/gcbrun request_pr_exp.py -n dg-g15 -m vertex_ai_gemini-1-5 -f

**Comment by DonggeLiu** [comment]
/gcbrun request_pr_exp.py -n dg-g15 -m vertex_ai_gemini-1-5 -f

**Comment by DonggeLiu** [comment]
The results look better and free from `Gemini-1.5`-specific issues now.

**Pull request merged**